cpp/src/arrow/util/bitmap_ops.h

```diff
@@ -124,7 +124,8 @@ Result<std::shared_ptr<Buffer>> BitmapAnd(MemoryPool* pool, const uint8_t* left,
 /// the results in out starting at the given bit-offset.
 ARROW_EXPORT
 void BitmapAnd(const uint8_t* left, int64_t left_offset, const uint8_t* right,
-               int64_t right_offset, int64_t length, int64_t out_offset, uint8_t* out);
+               int64_t right_offset, int64_t length, int64_t out_offset, uint8_t* out,
+               int64_t* const out_pop_count);
```

Comment:
remove `const`, and make `nullptr` the default value as commented.

Reply:
I added this `const` qualifier to indicate pointer wouldn't change.

Reply:
`const` parameter may be useful in definition, but putting it in declaration is a bit weird IMO.
Another reason is just to be consistent with existing parameters.

Reply:
Sure